.github/workflows/build.yml

```diff
@@ -40,6 +40,8 @@ jobs:
       run: cargo build --release
     - name: Run tests
       run: cargo test --release --workspace
+    - name: Run functional tests
+      run: cargo test --release -p mintlayer-test --test functional -- --ignored
```

Comment:
Having this in the CI takes the "completion" time to ~20 mins for the CI. Is that a good idea? I genuinely don't know.

Reply:
I like the idea of having it in the CI to catch issues

Reply:
There's no question that running functional tests in CI is required. In fact, this is the right place to run them, before merging and breaking the node because of some new code.